src/com/vogon101/Graphics/Controls/InputManager.scala

```diff
@@ -0,0 +1,51 @@
+package com.vogon101.Graphics.Controls
+
+import java.util
+
+import com.colofabrix.scala.geometry.shapes.Box
+import com.colofabrix.scala.math.{CartesianCoord, Vector2D}
+import org.lwjgl.input.Mouse
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Created by Freddie on 19/05/2015.
+ *
+ * This class contains all the buttons for controlling the game
+ * Has an ArrayBuffer of ButtonControl to contain the buttons, the
+ * update() method should be called once per step to check for any
+ * clicks. If any clicks are found within a button, that button's
+ * action will be called
+ */
+class InputManager {
+
+  private val _buttons: ArrayBuffer[ButtonControl] = new ArrayBuffer[ButtonControl]()
```

Comment:
Can't understand why you used a mutable. This is supposed to contains the buttons of the graphic that should be knowledge of the WorldRenderer that instruct this input manager